### PR TITLE
Thyra:  Remove Unused Parameter Warnings

### PR DESCRIPTION
**Pull request by jmgate**
@trilinos/thyra 

## Description
Comment out parameter names in function definitions to avoid
`-Wunused-parameter` warnings.

## Motivation and Context
Clean build.

### CODE REVIEW

File: packages/thyra/core/src/support/nonlinear/model_evaluator/client_support/Thyra_DirectionalFiniteDiffCalculator_def.hpp [hunk 1]
```diff
@@ -160,6 +160,9 @@ class OutArgsCreator : public StateFuncModelEvaluatorBase<Scalar>
         << ", the space p_space("<<l<<") must be in-core so that they can"
         " act as the domain space for the multi-vector derivative!"
         );
+#else
+      (void)model;
```
Comment (mhoemmen) [inline packages/thyra/core/src/support/nonlinear/model_evaluator/client_support/Thyra_DirectionalFiniteDiffCalculator_def.hpp]: Casting to `void` doesn't always fix "unused variable" compiler warnings, but it's a start.

**Comment by bartlettroscoe** [comment]
FYI: Looks like this failed due to random build errors, see https://github.com/trilinos/Trilinos/issues/3276#issuecomment-477570548.

I added the `AT: RETEST` label in hopes we get lucky the next time.

File: packages/thyra/core/src/support/nonlinear/model_evaluator/client_support/Thyra_DirectionalFiniteDiffCalculator_def.hpp (hunk 1 above)
Comment (jmgate) [reply packages/thyra/core/src/support/nonlinear/model_evaluator/client_support/Thyra_DirectionalFiniteDiffCalculator_def.hpp]: Hmmm, I've seen it work any time I've tried it, but perhaps that's just for gcc.  Is there a better way?

**Comment by jmgate** [comment]
Same random build failures we've been seeing lately.

**Pull request merged**